src/test/random/test_random.c

```diff
@@ -3,73 +3,134 @@
  * See LICENSE for licensing information
  */
 
+#include <assert.h>
 #include <fcntl.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
+#include <sys/syscall.h>
 #include <sys/types.h>
 #include <unistd.h>
 
-#define LOW_THRESH 0.1f
-#define HIGH_THRESH 0.9f
+// The number of random values to generate with each method.
+#define RGENLEN 100
+// The number of buckets to use when checking random value distribution.
+#define BUCKETLEN 10
 
-static int _test_dev_urandom() {
-    const char* path = "/dev/urandom";
-    uint numLow = 0, numHigh = 0;
+static int _check_randomness(double* fracs, int len) {
```

Comment:
Seems worth briefly commenting on the methodology here. It looks like the idea is that this is basically a fast "plausibly random" test suitable for testing that the plumbing isn't totally broken, but isn't trying to really evaluate the underlying RNG?

Reply:
Comment added.

(You're right, we're just testing the "plumbing" and not the quality of the underlying RNGs. Another useful test here is for determinism, but that is currently tested as part of the determinism test case (though we may decide to move it here later).)